src/it/scala/MessageRecoveryISpec.scala

```diff
@@ -0,0 +1,131 @@
+
+  implicit override val patienceConfig: PatienceConfig =
+    PatienceConfig(timeout = scaled(Span(60, Seconds)), interval = scaled(Span(5, Millis)))
+
+  private val requestQueueName = "request-general"
+  private val replyQueueName = "omega-editorial-web-application-instance-1"
+  private val sqsHostName = "localhost"
+  private val sqsPort = 9324
+
+  private val messageId = Version1UUID.generate()
+
+  var replyMessageText: Option[String] = None
+  var replyMessageId: Option[String] = None
+  var tempMsgDir: Option[String] = None
+  var apiService: Option[ApiService] = None
+
+  private val jmsClient = simpleQueueService.makeJmsClient[IO](
+    Config(
+      endpoint = simpleQueueService.Endpoint(Some(DirectAddress(HTTP, sqsHostName, Some(sqsPort))), "elasticmq"),
+      credentials = Some(Credentials("x", "x")),
+      clientId = simpleQueueService.ClientId("ctd-omega-services"),
+      None
+    )
+  )
+
+  private def readTextMessage(jmsMessage: JmsMessage): IO[Unit] = {
+    replyMessageId = jmsMessage.getJMSCorrelationId
+    jmsMessage.asTextF[IO].attempt.map {
+      case Right(text) =>
+        replyMessageText = Some(text)
+      case Left(e) => fail(s"Unable to read message contents due to ${e.getMessage}")
+    }
+  }
+
+  override protected def beforeAll(): Unit = {
+    // load messages to disk
+    val path = writeMessageFile(new LocalMessage(messageId, "Test World!", None, None, None, None, None, None, None))
+    tempMsgDir = Some(path.getParent.toString)
+    apiService = Some(
+      new ApiService(
+        ServiceConfig(
+          tempMessageDir = tempMsgDir.get,
+          maxConsumers = 1,
+          maxProducers = 1,
+          maxDispatchers = 1,
+          maxLocalQueueSize = 1,
+          requestQueue = requestQueueName,
+          replyQueue = replyQueueName
+        )
+      )
+    )
+
+    val consumerRes = for {
+      client <- jmsClient
+      consumer <-
+        client.createAutoAcknowledgerConsumer(QueueName(replyQueueName), 1, 100.millis)
+      _ <- Resource.eval(consumer.handle { (jmsMessage, _) =>
+             for {
+               _ <- readTextMessage(jmsMessage)
+             } yield AutoAckAction.noOp
+           })
+    } yield consumer
+    consumerRes.useForever.unsafeToFuture()
+    apiService.get.start.unsafeToFuture()
+    ()
+  }
+  override def afterAll(): Unit =
+    apiService.get.stop().unsafeToFuture()
+
+  "The Message Recovery API" - {
+
+    "runs the recovery service and removes the message from the message store" in {
+      val messageStoreFolder = Paths.get(apiService.get.config.tempMessageDir)
+      val localMessageStore = new LocalMessageStore(messageStoreFolder)
+      eventually {
+        localMessageStore.readMessage(messageId).asserting(_.failure.exception mustBe a[NoSuchFileException]) *>
+          IO.pure(replyMessageText).asserting(_ mustBe Some("The Echo Service says: Test World!"))
```

Comment:
You could put the message text in a `val` to show that the reply is the same thing. e.g. `s"The Echo Service says: $testMessage"`

Reply:
Done